extensions/wikia/ContributionPrototype/ContributionPrototype.setup.php

```diff
@@ -0,0 +1,16 @@
+<?php
+spl_autoload_register(function($class) {
+	$prefix = 'ContributionPrototype\\';
+	$len = strlen($prefix);
+	if (strncmp($prefix, $class, $len) !== 0) {
+		return;
+	}
+
+	$file = __DIR__.'/'.str_replace('\\', '/', substr($class, $len)).'.php';
+	if (file_exists($file)) {
+		require $file;
+	}
+});
+
+// take over actions
+$wgActions['view'] = ContributionPrototype\CPViewAction::class;
```

Comment:
Yah. There are several actions. `view` refers to the action of article viewing, with the default being `null` which means to use `ViewAction`. We're hijacking that :)